opm/output/Wells.hpp

```diff
@@ -88,7 +89,32 @@ namespace Opm {
         std::map< Completion::active_index, Completion > completions;
     };
 
-    typedef std::map< std::string, Well > Wells;
+    struct Wells {
+        using value_type = std::map< std::string, Well >::value_type;
+        using iterator = std::map< std::string, Well >::iterator;
+
+        inline Well& operator[]( const std::string& );
+        inline Well& at( const std::string& );
+        inline const Well& at( const std::string& ) const;
+        template< typename... Args >
+        inline std::pair< iterator, bool > emplace( Args&&... );
+
+        inline Wells() = default;
+        inline Wells( std::initializer_list< value_type > );
+        inline Wells( std::initializer_list< value_type >,
+                      std::vector< double > bhp,
+                      std::vector< double > temperature,
+                      std::vector< double > wellrates,
+                      std::vector< double > perf_pressure,
+                      std::vector< double > perf_rates );
+
+        std::map< std::string, Well > wells;
+        std::vector< double > bhp;
+        std::vector< double > temperature;
```

Comment:
The temperature variable is orders of magnitude less relevant than the others. In most(?) of the cases it will only have a fixed dummy value - if not downright removed it should at least be removed to the end of the argument list.